### PR TITLE
Reflection: Fix null as first return type

### PR DESCRIPTION
**Pull request by theCapypara**
### Description (*)
This fixes null as the first return type for methods used in the service layer.

The examples given in #25656 are now working. Additionally the Swagger schema generation no longer fails with the exception ```The "" data type isn't declared. Verify the type and try again.```.

~If null is the only return type, a new exception will be thrown.~ It's not, see comments below

### Fixed Issues
1. magento/magento2#25656: M2.3.2 : Nullable getters in Service Contracts will throw a reflection error when used in the web API

### Manual testing scenarios (*)
1. Create a simple REST WebAPI endpoint, with a getter Method as described in #25656 
2. Try to open the Swagger documentation (should work now)
3. Try to execute the API endpoint (should work now)

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by theCapypara** [comment]
I had to remove the exception I added to catch cases, where only null is defined as a return type for methods, which isn't supported by the code resolving type instances of return types.

The reason for this is, that the reflection logic also seems to be used during integration tests, and there are methods used there which actually return null. There are no errors being thrown there, so assume in those cases null is actually supported as a return type by the reflection logic. For reference, here are the logs of the integration builds:

```
11:34:01 [Integration-7-ce] In TypeProcessor.php line 297:
11:34:01 [Integration-7-ce]
11:34:01 [Integration-7-ce]   [InvalidArgumentException]
11:34:01 [Integration-7-ce]   No valid return type for Magento\TestModuleMysqlMq\Model\DataObjectReposito
11:34:01 [Integration-7-ce]   ry::delayedOperation() specified. Verify the return type and try again.
11:34:01 [Integration-7-ce]
11:34:01 [Integration-7-ce]
11:34:01 [Integration-7-ce] Exception trace:
11:34:01 [Integration-7-ce]  () at /var/www/html/lib/internal/Magento/Framework/Reflection/TypeProcessor.php:297
11:34:01 [Integration-7-ce]  Magento\Framework\Reflection\TypeProcessor->getGetterReturnType() at /var/www/html/lib/internal/Magento/Framework/Reflection/MethodsMap.php:173
11:34:01 [Integration-7-ce]  Magento\Framework\Reflection\MethodsMap->getMethodMapViaReflection() at /var/www/html/lib/internal/Magento/Framework/Reflection/MethodsMap.php:108
11:34:01 [Integration-7-ce]  Magento\Framework\Reflection\MethodsMap->getMethodsMap() at /var/www/html/lib/internal/Magento/Framework/Reflection/MethodsMap.php:81
```

This means methods that only have null as a return type will still throw the same weird exceptions mentioned in the issue (when used via webapi), but the integration tests are working again.

Methods that can return null AND a valid type are now working correctly, regardless of order.

File: lib/internal/Magento/Framework/Reflection/TypeProcessor.php [hunk 1]
```diff
@@ -286,7 +286,14 @@ public function getGetterReturnType($methodReflection)
     {
         $returnAnnotation = $this->getMethodReturnAnnotation($methodReflection);
         $types = $returnAnnotation->getTypes();
-        $returnType = current($types);
+        $returnType = null;
+        foreach ($types as $type) {
+            if ($type != 'null') {
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/Reflection/TypeProcessor.php]:
```suggestion
            if ($type !== 'null') {
```

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6920](https://jira.corp.magento.com/browse/ENGCOM-6920) has been created to process this Pull Request
:eight_spoked_asterisk: @ihor-sviziev, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed
